roles/test-utils/mining-device/src/lib/mod.rs

```diff
@@ -0,0 +1,711 @@
+                    .unwrap();
+                notify_changes_to_mining_thread.should_send = false;
+            };
+            match next {
+                SendTo::RelayNewMessageToRemote(_, m) => {
+                    let sv2_frame: StdFrame = MiningDeviceMessages::Mining(m).try_into().unwrap();
+                    let either_frame: EitherFrame = sv2_frame.into();
+                    sender.send(either_frame).await.unwrap();
+                }
+                SendTo::None(_) => (),
+                _ => panic!(),
+            }
+        }
+    }
+
+    async fn send_share(
+        self_mutex: Arc<Mutex<Self>>,
+        nonce: u32,
+        job_id: u32,
+        version: u32,
+        ntime: u32,
+    ) {
+        let share =
+            MiningDeviceMessages::Mining(Mining::SubmitSharesStandard(SubmitSharesStandard {
+                channel_id: self_mutex.safe_lock(|s| s.channel_id.unwrap()).unwrap(),
+                sequence_number: self_mutex.safe_lock(|s| s.sequence_numbers.next()).unwrap(),
+                job_id,
+                nonce,
+                ntime,
+                version,
+            }));
+        let frame: StdFrame = share.try_into().unwrap();
+        let sender = self_mutex.safe_lock(|s| s.sender.clone()).unwrap();
+        sender.send(frame.into()).await.unwrap();
+    }
+}
+
+impl IsUpstream<(), NullDownstreamMiningSelector> for Device {
+    fn get_version(&self) -> u16 {
+        todo!()
+    }
+
+    fn get_flags(&self) -> u32 {
+        todo!()
+    }
+
+    fn get_supported_protocols(&self) -> Vec<Protocol> {
+        todo!()
+    }
+
+    fn get_id(&self) -> u32 {
+        todo!()
+    }
+
+    fn get_mapper(&mut self) -> Option<&mut roles_logic_sv2::common_properties::RequestIdMapper> {
+        todo!()
+    }
+
+    fn get_remote_selector(&mut self) -> &mut NullDownstreamMiningSelector {
+        todo!()
+    }
+}
+
+impl IsMiningUpstream<(), NullDownstreamMiningSelector> for Device {
+    fn total_hash_rate(&self) -> u64 {
+        todo!()
+    }
+
+    fn add_hash_rate(&mut self, _to_add: u64) {
+        todo!()
+    }
+    fn get_opened_channels(
+        &mut self,
+    ) -> &mut Vec<roles_logic_sv2::common_properties::UpstreamChannel> {
+        todo!()
+    }
+
+    fn update_channels(&mut self, _: roles_logic_sv2::common_properties::UpstreamChannel) {
+        todo!()
+    }
+}
+
+impl ParseUpstreamMiningMessages<(), NullDownstreamMiningSelector, NoRouting> for Device {
+    fn get_channel_type(&self) -> SupportedChannelTypes {
+        SupportedChannelTypes::Standard
+    }
+
+    fn is_work_selection_enabled(&self) -> bool {
+        false
+    }
+
+    fn handle_open_standard_mining_channel_success(
+        &mut self,
+        m: OpenStandardMiningChannelSuccess,
+        _: Option<std::sync::Arc<Mutex<()>>>,
+    ) -> Result<SendTo<()>, Error> {
+        self.channel_opened = true;
+        self.channel_id = Some(m.channel_id);
+        let req_id = m.get_request_id_as_u32();
```

Comment:
The `cargo clippy` CI fails on this line with:

```sh
warning: unused variable: `req_id`
   --> test-utils/mining-device/src/lib/mod.rs:389:13
    |
389 |         let req_id = m.get_request_id_as_u32();
    |             ^^^^^^ help: if this is intentional, prefix it with an underscore: `_req_id`
    |
    = note: `#[warn(unused_variables)]` on by default
```

The reason it fails is because the `main` branch immediately uses this `req_id` in an `info!` log on [L438](https://github.com/stratum-mining/stratum/blob/ab0ddd625706698c47d44b9b4e4f985c39be18db/roles/test-utils/mining-device/src/main.rs#L438) with:

```rs
        info!(
            "MINING DEVICE: channel opened with: group id {}, channel id {}, request id {}",
            m.group_channel_id, m.channel_id, req_id
        );
```

Why did we remove this log? Are there any other logs that got removed? Are there any other changes in the code in this PR beyond breaking up the logic into the `mod.rs` file?

Reply:
@jbesraa we need to make sure that this PR is strictly moving lines of code to different files but without losing logic, which seems to be the root cause of this